util/src/main/java/io/kubernetes/client/util/ClientBuilder.java

```diff
@@ -0,0 +1,242 @@
+
+	private static final Logger log = Logger.getLogger(Config.class);
+
+	public String getUserName() {
+		return userName;
+	}
+
+	public ClientBuilder setUserName(String userName) {
+		this.userName = userName;
+		return this;
+	}
+
+	public String getPassword() {
+		return password;
+	}
+
+	public ClientBuilder setPassword(String password) {
+		this.password = password;
+		return this;
+	}
+
+	public String getApiKey() {
+		return apiKey;
+	}
+
+	public ClientBuilder setApiKey(String apiKey) {
+		this.apiKey = apiKey;
+		return this;
+	}
+
+	public String getBasePath() {
+		return basePath;
+	}
+
+	public ClientBuilder setBasePath(String basePath) {
+		this.basePath = basePath;
+		return this;
+	}
+
+	public File getCertificateAuthorityFile() {
+		return certificateAuthorityFile;
+	}
+
+	public ClientBuilder setCertificateAuthority(File certificateAuthorityFile) {
+		this.certificateAuthorityFile = certificateAuthorityFile;
+		this.verifyingSsl = true;
+		return this;
+	}
+
+	public String getCertificateAuthorityData() {
+		return certificateAuthorityData;
+	}
+
+	public ClientBuilder setCertificateAuthority(String certificateAuthorityData) {
+		this.certificateAuthorityData = certificateAuthorityData;
+		this.verifyingSsl = true;
+		return this;
+	}
+
+	public ClientBuilder setClusterMode() throws IOException {
+		this.client = Config.fromCluster();
+		return this;
+	}
+
+	public ClientBuilder setKubeConfig(KubeConfig config) {
+		this.kubeConfig = config;
+		if( this.kubeConfig !=null) {
+			this.client = Config.fromConfig(this.kubeConfig);
+		}
+		return this;
+	}
+
+	public ClientBuilder setDefaultKubeConfigMode() throws FileNotFoundException {
+		this.client = Config.fromConfig(KubeConfig.loadDefaultKubeConfig());
+		return this;
+	}
+
+	public ClientBuilder setKubeConfig(File kubeFile) throws FileNotFoundException {
+		this.kubeConfig = KubeConfig.loadKubeConfig(new FileReader(kubeFile));
+		return this;
+	}
+
+	public ClientBuilder setKubeConfig(Reader input) {
+		this.kubeConfig = KubeConfig.loadKubeConfig(input);
+		return this;
+	}
+
+	public ClientBuilder setKubeConfig(InputStream stream) {
+		this.kubeConfig = KubeConfig.loadKubeConfig(new InputStreamReader(stream));
+		return this;
+	}
+
+	public KeyManager[] getKeyMgrs() {
+		return keyMgrs;
+	}
+
+	public ClientBuilder setKeyMgrs(KeyManager[] keyMgrs) {
+		this.keyMgrs = keyMgrs;
+		return this;
+	}
+
+	public boolean isVerifyingSsl() {
+		return verifyingSsl;
+	}
+
+	public ClientBuilder setVerifyingSsl(boolean verifyingSsl) {
+		this.verifyingSsl = verifyingSsl;
+		return this;
+	}
+
+
+	public ClientBuilder setDefaultClientMode() throws IOException {
+		client = Config.defaultClient();
+		return this;
+	}
+
+	public String getApiKeyPrefix() {
+		return apiKeyPrefix;
+	}
+
+	public ClientBuilder setApiKeyPrefix(String apiKeyPrefix) {
+		this.apiKeyPrefix = apiKeyPrefix;
+		return this;
+	}
+
+	public ApiClient build() throws FileNotFoundException {
```

Comment:
I'd like to see this get to a point where it doesn't throw `FileNotFoundException` I think that means moving the cert loading to the setter function.